frontend/models/product-list/server.ts

```diff
@@ -139,6 +148,21 @@ export function getProductListType(
    }
 }
 
+function filterDevicesWithNoProducts(
+   productListChildren: ProductListChild[],
+   devicesWithProducts: string[] | null
+) {
+   if (devicesWithProducts == null) {
+      return productListChildren;
+   }
+   if (devicesWithProducts.length === 0) {
+      return [];
+   }
+   return productListChildren.filter((child) =>
+      devicesWithProducts.some((device) => device === child.deviceTitle)
```

Comment:
Let's use a `Map` so we can do a faster check than a linear array search.

Reply:
https://github.com/iFixit/react-commerce/pull/1252/commits/93f54047b62476cd6da28dedef209128678710bd